vncdotool/loggingproxy.py

```diff
@@ -55,11 +57,17 @@ def _handle_version(self):
                 self._handler = self._handle_clientInit, 1
         elif version in ('007', '008'):
             # XXX send security v3.7+
-            self._handler = self._handle_security, 1
+            self._handler = self._handle_security, 2
```

Comment:
How did this work before if it was chopping off the wrong number of bytes? Wouldn't it break on the next message? Is 2 definitely correct?

Reply:
The first two questions you asked here i can answer with ease...
> How did this work before if it was chopping off the wrong number of bytes? Wouldn't it break on the next message?
`_handle_security` did not do any thing with the data before. If `vnclog` connected to any server with RFB version 3.7 or 3.8, nothing would ever happen because they havn't been implemented in the logging proxy.

The 3rd question however is a bit more complicated.
The proxy server does not receive all the data packages and thus it does not follow the RFB protocol completely, nor should it. It's just there to capture data and should not respond to it.

The RFB protocol requires the client and server to negotiate for a [SecurityType](https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#security). This is done (in version 3.7+) by sending 1 byte informing the client that X number of SecurityTypes is supported by the server followed by X bytes describing the supported SecurityTypes. But all of this has all ready been handled by the client.

The first byte is the SecurityType that the client has chosen.
The second byte is the shared-flag and should be handled by [ClientInit](https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#clientinit), but I have not been able to find an easy solution to pass it to `_handle_clientInit` without getting an error: `builtins.AttributeError: 'NullTransport' object has no attribute 'addressFamily'`.

Reply:
Fixed by fdb1bf5d